documentation/state-diagrams/transfer-internal-states.plantuml

```diff
@@ -0,0 +1,75 @@
+@startuml
+
+state RECEIVED {
+  state RECEIVED_PREPARE {
+  }
+}
+
+state RESERVED_ {
+  state RESERVED {
+  }
+  state RESERVED_FORWARDED {
+  }
+  state RECEIVED_FULFIL {
+  }
+  state RECEIVED_FULFIL_DEPENDENT {
+  }
+  state RESERVED_TIMEOUT {
+  }
+  state RECEIVED_REJECT {
+  }
+  state RECEIVED_ERROR {
+  }
+}
+
+state COMMITTED {
+}
+
+state ABORTED {
+  state ABORTED_ERROR {
+  }
+  state ABORTED_REJECTED {
+  }
+  state EXPIRED_PREPARED {
+  }
+  state EXPIRED_RESERVED {
+  }
+  state FAILED {
+  }
+  state INVALID {
+  }
+}
+
+RECEIVED_FULFIL_DEPENDENT : only FX-transfer
+RECEIVED_FULFIL : only transfer
+
+[*] --> RECEIVED_PREPARE : Transfer Prepare Request [Prepare handler] \n <i>(validation & dupl.check passed)</i>
+[*] --> INVALID : Validation failed \n [Prepare handler]
+RECEIVED_PREPARE --> RESERVED : [Position handler]: Liquidity check passed, \n funds reserved
+RESERVED --> RECEIVED_REJECT : Reject callback from Payee with status "ABORTED"
+RECEIVED_PREPARE --> RECEIVED_ERROR : Transfer Error callback from Payee
```

Comment:
how can a Payee FSP respond to something thats not reserved at the switch and forwarded to it (the Payee FSP)? As in, if the Payee is able to reject it, it should've received it from the switch, where it should've been reserved. Isn't it?

Reply:
:thinking: :thinking: :thinking: 

I believe if a fxTransfer is reserved and a subsequent transfer is prepared and stuck in `RECEIVED_PREPARE` (example transfer position handler hangs). If the FXP returns an `/error` the handlers will abort both the fxTransfer and transfer thus performing the state change for the transfer `RECEIVED_PREPARE --> RECEIVED_ERROR`.

I believe that is the hypothetical case :thinking: 

@geka-evk 
@vijayg10 

Anything to add or is this a mistake.

Reply:
Thanks Kevin. Even in that case, saying "Reject callback from Payee with status "ABORTED" doesn't seem apt; maybe a concise version of what you said above..

Reply:
Talked it out with Vijay. This is just a mistake. Removed.

Reply:
Thanks @kleyow !